packages/sync-actions/src/utils/copy-empty-array-props.js

```diff
@@ -7,17 +32,8 @@
  */
 
 export default function copyEmptyArrayProps(oldObj, newObj) {
-  Object.entries(oldObj).forEach(([key, value]) => {
-    if (Array.isArray(value)) {
-      const foundKey = Object.keys(newObj).findIndex(
-        (newObjKey) => JSON.stringify(newObjKey) === JSON.stringify(key)
-      )
-
-      if (foundKey === -1) {
-        newObj[key] = [] // eslint-disable-line no-param-reassign
-      }
-    }
-  })
+  const oldObjCopy = deepCopyObj(oldObj)
+  const newObjCopy = deepCopyObj(newObj)
 
-  return [oldObj, newObj]
+  return copyEmptyArrayPropsRecursion(oldObjCopy, newObjCopy)
```

Comment:
You now solve the issue by copying the input parameters, which obviously works. But if I have to fix another bug in this module in the future, I will still have to understand that the `copyEmptyArrayPropsRecursion` has the side effect that it modifies its input parameter - which I consider as a bad practice. How difficult would it to be to fix this function to not mutate the parameters?

Reply:
@MarkusAzer See PR #1526 😄 😉

Reply:
@katmatt i really love the test :), but the coping nested object didnt added!
could you take alook on the new changes?!

Reply:
@MarkusAzer I would prefer if you would review my PR together with @daern91 i think it makes more sense to use `Object.fromEntries` in this case, because it's the inverse function of `Object.entries` and easier to understand then `reduce`. 
If we merged that PR to master you can merge master to this branch. And then you can add the recursion and update the test to have the recursion covered too. What do you think, @MarkusAzer ? Sounds like a plan?

Reply:
Yep 😎 plan